featurehub-javascript-client-sdk/app/feature_state_holders.ts

```diff
@@ -211,14 +215,14 @@ export class FeatureStateBaseHolder implements FeatureStateHolder {
       const matched = this._repo.apply(featureState.strategies, this._key, featureState.id, this._ctx);
 
       if (matched.matched) {
-        return this._castType(type, matched.value);
+        return this._castType(type, matched.value, parseJson);
       }
     }
 
     return featureState?.value;
```

Comment:
this will mean if the value isn't intercepted, the json is not parsed, it will be returned as a raw string, you may wish to add a test to confirm this...